modules/host/negotiateformcontract.go

```diff
@@ -166,7 +148,15 @@ func (h *Host) managedFinalizeContract(builder modules.TransactionBuilder, rente
 		LockedCollateral:     hostPortion,
 		OriginTransactionSet: fullTxnSet,
 	}
+	err = h.lockStorageObligation(so)
+	if err != nil {
+		return nil, err
+	}
 	err = h.addStorageObligation(so)
+	err2 := h.unlockStorageObligation(so)
+	if err2 != nil {
+		return nil, err2
+	}
```

Comment:
This was code that I wrote, so it's my fault, but this cluster is pretty messy. Maybe there's a better organization? At the very least, err2->err and err->addObligationErr or something